### PR TITLE
Tmz 3.2.0 enhance gcw script

### PR DESCRIPTION
**Pull request by t-m-z**
- add three string functions:
   + padleft
   + padright
   + padcenter
- add crypto functions
   + bifid
   + trifid
- add function
   + textanalysis
- add mapcode

### CODE REVIEW

File: lib/tools/coords/_common/formats/mapcode/logic/external_libs/mapcode.dart [hunk 1]
```diff
@@ -3965,4 +3965,7 @@ class McInfo {
   String territoryAlphaCode = '';
   String fullmapcode = '';
   int territoryNumber = 0;
+
+  McInfo();
```
Comment (Mike-3) [inline lib/tools/coords/_common/formats/mapcode/logic/external_libs/mapcode.dart:3969]: Neccesary ?

File: lib/tools/crypto_and_encodings/bifid/logic/bifid.dart [hunk 2]
```diff
@@ -50,7 +50,8 @@ BifidOutput decryptBifid(String input, String key,
     {PolybiosMode mode = PolybiosMode.AZ09,
     String? alphabet,
     AlphabetModificationMode alphabetMode = AlphabetModificationMode.J_TO_I}) {
-  int dim = key.length;
+
+int dim = key.length;
```
Comment (Mike-3) [inline lib/tools/crypto_and_encodings/bifid/logic/bifid.dart:54]: format code

**Review by Mike-3 — CHANGES_REQUESTED** [review]

File: lib/tools/coords/_common/formats/mapcode/logic/external_libs/mapcode.dart (hunk 1 above)
Comment (t-m-z) [reply lib/tools/coords/_common/formats/mapcode/logic/external_libs/mapcode.dart:3969]: no idea where this came from. will check/compare to 3.2.0 and probably remove

File: lib/tools/crypto_and_encodings/bifid/logic/bifid.dart (hunk 2 above)
Comment (t-m-z) [reply lib/tools/crypto_and_encodings/bifid/logic/bifid.dart:54]: done

**Pull request merged**